tests/h/search/query_test.py

```diff
@@ -418,9 +418,9 @@ def test_filters_on_whole_url(self, Annotation, get_search):
     def test_filter_matches_invalid_uri(self, Annotation, get_search):
         search = get_search()
         Annotation(target_uri="https://bar.com")
-        expected_ids = [Annotation(target_uri="invalid-uri").id]
+        expected_ids = [Annotation(target_uri="https://-invalid-uri").id]
```

Comment:
Yeah, the only case I can think of is you can make an annotation via the api and pass it whatever url you want and since the schema doesn't actually validate it except for being a string you could specify invalid uri/urls in there. We already have scheme requirements for wildcard uri's thought. Perhaps we should validate uri and url's to conform to some uri/url standard in the schema. The only validation I see is in the uri normalization which I wouldn't _really_ call validation since it just spits back out the invalid uri's. Previous behavior was that whatever uri was passed in (if invalid) was passed directly to search. I feel like this need a team level decision.

Reply:
see [slack convo](https://hypothes-is.slack.com/archives/C4K6M7P5E/p1544231948224800)

Reply:
Just a note, I think we could merge this PR w/o adding validation for uri/urls to have scheme's in the api. I don't see any searches being run via the api on non-schemed uri/urls and worst case it adds the http scheme if it's missing.